internal/primitive/query.go

```diff
@@ -170,6 +212,10 @@ func (e *Query) Run(ctx context.Context, input modconfig.Input) (*modconfig.Outp
 	}
 
 	if err = rows.Err(); err != nil {
+		// Check for context deadline exceeded error
+		if errors.Is(err, context.DeadlineExceeded) {
+			return nil, perr.TimeoutWithMessage("Query execution exceeded timeout")
+		}
 		return nil, err
```

Comment:
We haven't done this properly before, but can we wrap this error with perr.Internal please.